src/openforms/js/components/admin/form_design/variables/VariablesEditor.stories.js

```diff
@@ -403,6 +403,162 @@ export const WithObjectsAPIRegistrationBackends = {
+        initialValue: '',
+      },
+      {
+        form: 'http://localhost:8000/api/v2/forms/36612390',
+        formDefinition: 'unsaved-step',
+        name: 'Select boxes list',
+        key: 'selectBoxesList',
+        source: 'component',
+        prefillPlugin: '',
+        prefillAttribute: '',
+        prefillIdentifierRole: 'main',
+        dataFormat: undefined,
+        isSensitiveData: false,
+        serviceFetchConfiguration: undefined,
+        initialValue: [],
+      },
+      {
+        form: 'http://localhost:8000/api/v2/forms/36612390',
+        formDefinition: 'unsaved-step',
+        name: 'Select boxes object',
+        key: 'selectBoxesObj',
+        source: 'component',
+        prefillPlugin: '',
+        prefillAttribute: '',
+        prefillIdentifierRole: 'main',
+        dataFormat: undefined,
+        isSensitiveData: false,
+        serviceFetchConfiguration: undefined,
+        initialValue: [],
+      },
+    ],
+    availableComponents: {
+      formioComponent: {
+        key: 'formioComponent',
+        type: 'textfield',
+      },
+      selectBoxesList: {
+        type: 'selectboxes',
+        multiple: false,
+        key: 'selectBoxesList',
+      },
+      selectBoxesObj: {
+        type: 'selectboxes',
+        multiple: false,
+        key: 'selectBoxesObj',
+      },
+    },
+    registrationBackends: [
+      {
+        backend: 'objects_api',
+        key: 'objects_api_1',
+        name: 'Example Objects API reg.',
+        options: {
+          version: 2,
+          objectsApiGroup: 1,
+          objecttype: '2c77babf-a967-4057-9969-0200320d23f1',
+          objecttypeVersion: 2,
+          variablesMapping: [],
+          transformToList: ['selectBoxesList'],
+        },
+      },
+    ],
+  },
+  parameters: {
+    msw: {
+      handlers: {
+        objectTypeTargetPaths: [
+          mockTargetPathsPost({
+            object: [
+              {
+                targetPath: ['path', 'to.the', 'target'],
+                isRequired: false,
+                jsonSchema: {type: 'object'},
+              },
+            ],
+            array: [
+              {
+                targetPath: ['other', 'path'],
+                isRequired: false,
+                jsonSchema: {type: 'array'},
+              },
+            ],
+            string: [
+              {
+                targetPath: ['path'],
+                isRequired: false,
+                jsonSchema: {type: 'string'},
+              },
+            ],
+          }),
+        ],
+      },
+    },
+  },
+  play: async ({canvasElement, step}) => {
+    const canvas = within(canvasElement);
+    const editIcons = canvas.getAllByTitle('Registratie-instellingen bewerken');
+
+    await expect(editIcons).toHaveLength(3);
+
+    await step('Select boxes with transform to list', async () => {
+      await userEvent.click(editIcons[1]);
+
+      const checkboxes = await canvas.findAllByRole('checkbox');
```

Comment:
can you use a `findByLabelText` instead? I can't tell now what the intention is, and if we re-order checkboxes in the UI the tests will break, which won't happen if we grab them by label.

Reply:
I did it because this is the case in the other tests but I will change them yes. Should I do it for all the similar situations in the older tests?

Reply:
yeah, let's take the time to clean that up!

Reply:
Discussed this on slack, it will be tackled during the new re-design.